packages/client/src/store/CallState.ts

```diff
@@ -390,6 +396,10 @@ export class CallState {
     this.participantCount$ = duc(this.participantCountSubject);
     this.recording$ = duc(this.recordingSubject);
     this.transcribing$ = duc(this.transcribingSubject);
+    this.duration$ = duc(this.durationSubject);
+    setInterval(() => {
+      this.setDuration((d) => d + 1);
+    }, 1000);
```

Comment:
If we set an interval like this, we need to also clear it somewhere, otherwise it's a memory leak.

You can also consider setting interval in Rx (with `timer` or `interval`) instead.

Reply:
Good catch about the memory leak 👍 I wasn't able to find good place to unsubscribe to this `setInterval` (no such thing as destructors in js) so I opted for different solution, moving this out of the call state.

Also the new hook takes `callStarted` timestamp as well, so on rejoining call or other future call joiners will see the proper time duration of a call.

@oliverlaz @myandrienko can we proceed with this solution or do you have a better idea?

Reply:
@kristian-mkd, this should live in `callState` as otherwise, it would be unavailable for Plain-JS.
In our CC PR, I've introduced `CallState.#dispose()` method, you can add it here too, and I'll resolve the conflicts.
- https://github.com/GetStream/stream-video-js/pull/1508/files#diff-e07349e380403998673da01d8b5dc1ebc2e7955bd23a6cf82409c21b02d8dda9R494-R503

Additionally, the interval should start automatically when
- joined (init with `session.live_started_at`)
- call.session_started event is emitted
- call.session_ended should stop it

Reply:
@oliverlaz I have pushed the suggested changes. Please verify them, if all good I will update the docs.